modules/programs/granted.nix

```diff
@@ -0,0 +1,35 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.granted;
+  package = pkgs.granted;
+
+in {
+  meta.maintainers = [ hm.maintainers.wcarlsen ];
+
+  options.programs.granted = {
+    enable = mkEnableOption "granted";
+
+    enableZshIntegration = mkOption {
+      default = true;
+      type = types.bool;
+      description = ''
+        Whether to enable Zsh integration.
+      '';
+    };
+  };
+
+  config = let
+    manualShellProfile = ''
+      alias assume="source ${package}/bin/.assume-wrapped"
+      export GRANTED_ALIAS_CONFIGURED="true"
```

Comment:
Wouldn't this leave a stray `GRANTED_ALIAS_CONFIGURED` in the environment? Perhaps something like

``` patch
diff --git a/modules/programs/granted.nix b/modules/programs/granted.nix
index 1ead35eb..41de1402 100644
--- a/modules/programs/granted.nix
+++ b/modules/programs/granted.nix
@@ -22,14 +22,15 @@ in {
     };
   };
 
-  config = let
-    manualShellProfile = ''
-      alias assume="source ${package}/bin/.assume-wrapped"
-      export GRANTED_ALIAS_CONFIGURED="true"
-    '';
-  in mkIf cfg.enable {
+  config = mkIf cfg.enable {
     home.packages = [ package ];
 
-    programs.zsh.envExtra = mkIf cfg.enableZshIntegration manualShellProfile;
+    programs.zsh.shellAliases = {
+      granted = concatStringsSep ";" [
+        "GRANTED_ALIAS_CONFIGURED='true'"
+        "source ${package}/bin/.assume-wrapped"
+        "unset GRANTED_ALIAS_CONFIGURED"
+      ];
+    };
   };
 }
```

could work?

Reply:
I agree that this is a better solution than mine. I will test it out to see if it works.

Reply:
Tested and it works beautifully.

Reply:
Yeah found a bug with the above solution this morning. Any parameters passed to to assume won't be propagated correct.

Reply:
Yeah, that's a good point. I wonder if not a function would be more appropriate? :-D

``` zsh
programs.zsh.initExtra = ''
  function assume() {
    export GRANTED_ALIAS_CONFIGURED='true'
    source ${package}/bin/.assume-wrapped "$@"
    unset GRANTED_ALIAS_CONFIGURED
  }
'';
```

The [yazi module](https://github.com/nix-community/home-manager/blob/09587fbbc6a669f7725613e044c2577dc5d43ab5/modules/programs/yazi.nix) does something like this already.

Reply:
I think you are right. I will tweak and test ✌️

Reply:
I got to test it  and bundling the functionality into a function does the trick. Thanks for the valuable input @rycee .